### ISSUE TITLE
make pathlib.Path a first citizen by adding default converter

### ISSUE THREAD

**Issue by zhuoqiang** [epsy/clize#43]
if  object of type `pathlib.Path` is used as function argument, clize will complain:

```
ValueError: Cannot find value converter for default value PosixPath('/foo/bar'). Please specify one as an annotation.
If the default value's type should be used to convert the value, make sure it is decorated with clize.parser.value_converter()
```

how about add the converter in clize by default? Many of the standard lib in Python3 already changed to support pathlib

**Comment by epsy** [comment]
Good call: pathlib is included in Python, and as a bonus it is the way forward in dealing with paths in Python.

After briefly looking at the documentation, it looks like adding `pathlib.PurePath` as a key to the default converters table should be the best option. If a user wants to force a specific flavor, they must be able to specify one of the platform-specific types as converters.

**Comment by zhuoqiang** [comment]
It would be great if file path validation could also be specified in requirment, like `click`

http://click.pocoo.org/5/arguments/#file-path-arguments

**Comment by epsy** [comment]
`converters.file` already does path checking, but gives you a context manager to open the file instead of a path object. Do you have a use case for validating a path and returning the path?

**Comment by Shir0kamii** [comment]
This issue is solved with #47
